### PR TITLE
Yield event loop while receiving messages on rasa shell

### LINKED ISSUE #4630: Rasa shell hangs forever on `/stop`
**Rasa version**: 1.3.8

**Rasa X version** (if used & relevant):

**Python version**: 3.7.4

**Operating system** (windows, osx, ...): Fedora 30

**Issue**: The rasa shell hangs forever with the message `Killing Sanic server now`.

**Error (including full traceback)**:
```
(env) ✔︎ tmp (master) ✗ rasa shell
2019-10-17 16:37:42 INFO     root  - Connecting to channel 'cmdline' which was specified by the '--connector' argument. Any other channels will be ignored. To connect to all given channels, omit the '--connector' argument.
2019-10-17 16:37:42 INFO     root  - Starting Rasa server on http://localhost:5005
Bot loaded. Type a message and press enter (use '/stop' to exit): 
Your input ->  /stop                                                                                                                                   
2019-10-17 16:37:51 INFO     root  - Killing Sanic server now.
^C
```

**Command or request that led to error**:
Inside the rasa shell
```
/stop
```

**Content of configuration file (config.yml)** (if relevant):
```yml

```

**Content of domain file (domain.yml)** (if relevant):
```yml

```

I'd like to contribute if possible :smiley:

### PR DESCRIPTION
**Pull request by icaropires**
**Proposed changes**:
- Fixes #4630 
- Them problem was that the loop for getting messages wasn't yielding the event loop, not allowing finish tasks to be executed

**Status (please check what you already did)**:
- [x] made PR ready for code review
- [ ] added some tests for the functionality
- [ ] updated the documentation
- [x] updated the changelog
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa_nlu#code-style) for instructions)

### CODE REVIEW

File: CHANGELOG.rst [hunk 1]
```diff
@@ -24,6 +24,7 @@ Removed
 Fixed
 -----
 - ``MultiProjectImporter`` now imports files in the order of the import statements
+- Fixed server hanging forever on exitting rasa shell before first message
```
Comment (wochinge) [inline CHANGELOG.rst]:
```suggestion
- Fixed server hanging forever on exiting ``rasa shell`` before first message
```

File: rasa/core/channels/console.py [hunk 2]
```diff
@@ -104,15 +105,13 @@ def get_user_input(button_question: questionary.Question) -> Optional[Text]:
 
 async def record_messages(
     server_url=DEFAULT_SERVER_URL,
-    auth_token=None,
+    auth_token="",
```
Comment (wochinge) [inline rasa/core/channels/console.py]: 🚀

File: rasa/core/channels/console.py [hunk 3]
```diff
@@ -142,6 +142,7 @@ def get_user_input(button_question: questionary.Question) -> Optional[Text]:
                 button_question = print_bot_output(response)
 
         num_messages += 1
+        await asyncio.sleep(0)  # Yield event loop for others coroutines
```
Comment (wochinge) [inline rasa/core/channels/console.py]: Do you actually need this one? I tested it and it works also without?

File: rasa/core/channels/console.py [hunk 4]
```diff
@@ -122,6 +121,7 @@ def get_user_input(button_question: questionary.Question) -> Optional[Text]:
 
     num_messages = 0
     button_question = None
+    await asyncio.sleep(0.5)  # Wait for server beggining
```
Comment (wochinge) [inline rasa/core/channels/console.py]:
```suggestion
    await asyncio.sleep(0.5)  # Wait for server to start
```

File: rasa/core/channels/console.py (hunk 3 above)
Comment (icaropires) [reply rasa/core/channels/console.py]: @wochinge Indeed, I tested and it worked with the last commits :thinking: . But without it, no other actions can be executed on the event loop until the end of the `while` which can lead to other problems in the future. 

Should I remove it?

Comment (wochinge) [reply rasa/core/channels/console.py]: Let's keep it in. I think it makes sense to have it there.

**Comment by wochinge** [comment]
Nice! Will merge it when it passed the tests 🎉 Thanks a lot!

**Comment by icaropires** [comment]
> Nice! Will merge it when it passed the tests tada Thanks a lot!

Glad to help!

**Comment by wochinge** [comment]
merged 🎉

**Pull request merged**